iosApp/Survey/Sources/Presentation/Modules/ResetPassword/ResetPassword+DataSource.swift

```diff
@@ -0,0 +1,61 @@
+//  swiftlint:disable:this file_name
+//
+//  ResetPassword+DataSource.swift
+//  Survey
+//
+//  Created by Bliss on 24/11/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Combine
+import KMPNativeCoroutinesCombine
+import Shared
+
+extension ResetPasswordView {
+
+    final class DataSource: ObservableObject {
+
+        let viewModel: ResetPasswordViewModel
+        let notificationManager: NotificationManageable
+
+        @Published private(set) var viewState = ResetPasswordViewState()
+        @Published var showingErrorAlert = false
+        @Published var showingLoading = false
+        @Published var email: String = ""
+
+        private var cancellables = Set<AnyCancellable>()
+
+        init(
+            viewModel: ResetPasswordViewModel = KoinApplication.shared.inject(),
+            notificationManager: NotificationManageable = NotificationManager.current
+        ) {
+            self.viewModel = viewModel
+            self.notificationManager = notificationManager
+            createPublisher(for: viewModel.viewStateNative)
+                .catch { error -> Just<ResetPasswordViewState> in
+                    let resetPasswordViewState = ResetPasswordViewState(
+                        error: error.localizedDescription
+                    )
+                    return Just(resetPasswordViewState)
+                }
+                .receive(on: DispatchQueue.main)
+                .sink { [weak self] value in
+                    guard let self = self else { return }
+                    self.viewState = value
+                    self.showingErrorAlert = !value.error.string.isEmpty
+                    self.showingLoading = value.isLoading
+                    guard let notification = value.successNotification else { return }
+                    self.notificationManager.schedule(
+                        title: notification.title(),
+                        message: notification.message(),
+                        time: 0.1
+                    )
```

Comment:
This implementation block looks a bit huge with multiple tasks done here. Instead, it might be better to separate them into private functions per task. For example, we can create 2 private functions - `updateStates` and `scheduleNotification`. Then this block can be shortened to:
```suggestion
                    self.updateStates(with: value)
                    guard let notification = value.successNotification else { return }
                    self.scheduleNotification(notification)
```
What do you think? 🤔

Reply:
@minhnimble Fixed this, other comments are rebase issue.